### ISSUE TITLE
omego-for-ansible

### ISSUE THREAD

**Issue by joshmoore** [ome/omego#102]
cF:

 * https://github.com/openmicroscopy/omero-grid-docker/blob/dde3e6591b5628b6253fa720b7cb0790b8e3868c/omero-grid/Dockerfile#L11
 * https://github.com/IDR/pydoop-features/commit/a1a76e17b98cc9b53e20a19ed6e552af8775c7ef#diff-3254677a7917c6c01f55212f86c57fbfR30

To simplify adding requirements for the various `install` targets that `omego` provides, the following could be added:

 * dependency on ansible so that pip installing also installs ansible
 * a command `req` (or `deps` or `ansible` or `setup`) or a flag like `install --setup` could be added which:
   - generates a `requirements.yml`
   - runs `ansible-galaxy`
   - generates a playbook
   - (optionally) runs the playbook
 * _skip_ generating those files if they already exist

The requirements.yml file might look like:

```
# web
- src: openmicroscopy.ice
- src: openmicroscopy.omero-web-runtime
- src: openmicroscopy.nginx # optionally

# server
 - src: openmicroscopy.java
 - src: openmicroscopy.omero-server

# etc.
```

The playbook would largely might look like:

```
# server
- hosts: localhost:
   roles:
   - role: openmicroscopy.omero-server
```

The goal would be to reduce the number of steps and necessary files needed for the likes of docker to:

```
RUN pip install omego && omego install --with-playbook
```

**Comment by manics** [comment]
It sounds like this would introduce two-way coupling: omego requires ansible, but ansible OMERO roles require omego for downloading artefacts. I think it'd be better to split `omego` into:
- library of tools, e.g. for downloading, or database operations
- An application that contains the logic for installing OMERO, and makes use of the tools library.

**Comment by joshmoore** [comment]
Good point. How would you see breaking it apart? All in this repo? Multiple "omego plugins"?
